tests/console/upgrade_snapshots.pm

```diff
@@ -23,12 +23,17 @@ sub run {
     select_console 'root-console';
 
     my $waittime = 200 * get_var('TIMEOUT_SCALE', 1);
-    script_run("snapper list | tee /dev/$serialdev", 0);
+
+    # 'snapper list' will cost a lots of time to finish if btrfs-balance.sh was triggered
+    # To make it faster, we'd check if snapper have disable-used-space option.
+    # This is a workaround for bsc#1167353.
+    record_soft_failure('workaround for bsc#1167353');
```

Comment:
Please never add unconditional `record_soft_failure` calls as this can cause a lot of false positives. Better check for the error condition. In this case maybe you can check the process tree for balance running first. I also suggest to create a specific bug for snapper list, not just rely on the other generic bug report. If you can't reliably check the error condition better make the record_soft_failure a record_info instead

Reply:
I've checked the log and lemon agreed this is the same bug to #1167353. when btrfs-balanace was trigger we can't
check the ba;ance process easily. so I changed to record_info.  can you take a look again.

[ 2681.150519] btrfs-balance.sh[14426]: Data, single: total=40.01GiB, used=35.07GiB
[ 2681.150690] btrfs-balance.sh[14426]: System, DUP: total=32.00MiB, used=64.00KiB
[ 2681.150824] btrfs-balance.sh[14426]: Metadata, DUP: total=1.50GiB, used=822.00MiB
[ 2681.150956] btrfs-balance.sh[14426]: GlobalReserve, single: total=40.00MiB, used=0.00B
[ 2681.152682] btrfs-balance.sh[14426]: Filesystem Size Used Avail Use% Mounted on
[ 2681.152830] btrfs-balance.sh[14426]: /dev/vda3 63G 40G 22G 65% /
[ 2681.178847] BTRFS info (device vda3): balance: start -dusage=0